h2o-algos/src/main/java/hex/tree/SharedTree.java

```diff
@@ -500,6 +500,9 @@ protected final void scoreAndBuildTrees(boolean oob) {
                 _model._output.trimTo(bestNTrees);
                 _model.update(_job);
               }
+            } else {
```

Comment:
This is not going to work properly for `_parms._use_best_cv_iteration = false`.

IMHO this should just be `if (! _parms._is_cv_model)`

Reply:
The tests are passing for `_parms._use_best_cv_iteration = false`. Are you concern  about some race condition since CV handling `ntrees` parameter in `ModelTrainingCoordinator`?

I have implemented your  `!_parms._is_cv_model` and update the tests to run also for `_use_best_cv_iteration = false`.

Reply:
The tests are not going to fail because for CV models the actual value of ntrees is already recorded here: https://github.com/h2oai/h2o-3/blob/master/h2o-algos/src/main/java/hex/tree/SharedTree.java#L1200

My comment is concerning local consistency of the code, the IF statement should not be thrown off by _use_best_cv_iteration.

Reply:
Aaa yes, I undestand. Scenario for `_parms._is_cv_model=True` and `_parms._use_best_cv_iteration=False` is already handled [here](https://github.com/h2oai/h2o-3/blob/master/h2o-algos/src/main/java/hex/tree/SharedTree.java#L1200). So we need to cover only the `_parms._is_cv_model=False`..